### PR TITLE
fix friendbot sequence number issues

### PR DESCRIPTION
**Pull request by nikhilsaraf**
I've tested this locally and it refreshes the friendbot sequence number correctly now. One of the issues we had was that it was not correctly checking for the tx_bad_seq, which is fixed.

On another note, I’m seeing a lot of 504 gateway timeouts (error decoding horizon.Problem: EOF) and "400 Bad Request" as responses from horizon when I’m testing locally. I don't think this can be fixed since we naturally want to push multiple txns per ledger with friendbot and cannot preempt these responses (at least not in a way that I'm currently aware). We are seeing this now as opposed to earlier because this is the first time friendbot is hitting the horizon server directly as opposed to going through the txSub sub-system.

@nullstyle

### CODE REVIEW

File: services/friendbot/internal/friendbot.go [hunk 1]
```diff
@@ -17,64 +18,81 @@ type Bot struct {
...
+		channel <- err
+	} else {
+		channel <- result
 	}
-	return &result, err
 }
 
 func (bot *Bot) checkHandleBadSequence(err *horizon.Error) {
-	if err.Problem.Type != "tx_bad_seq" {
+	resCode, e := err.ResultCodes()
```
Comment (nikhilsaraf) [inline services/friendbot/internal/friendbot.go]: this line is the main fix for how we check for tx_bad_seq

File: services/friendbot/internal/friendbot.go [hunk 2]
```diff
@@ -111,13 +129,14 @@ func (bot *Bot) refreshSequence() error {
 		return err
 	}
 
-	seq, err := strconv.ParseInt(botAccount.Sequence, 10, 0)
+	seq, err := strconv.ParseInt(botAccount.Sequence, 10, 64)
```
Comment (nikhilsaraf) [inline services/friendbot/internal/friendbot.go]: force this to be 64 bit based on @bartekn's suggestion

File: services/friendbot/internal/friendbot.go [hunk 3]
```diff
@@ -17,64 +18,81 @@ type Bot struct {
...
+		return &tv, nil
+	case error:
+		return nil, tv
+	default:
+		return nil, fmt.Errorf("failed to submit async txn")
+	}
+}
+
+func (bot *Bot) lockedPay(channel chan interface{}, destAddress string) (bool, *horizon.TransactionSuccess, error) {
+	bot.lock.Lock()
```
Comment (nikhilsaraf) [inline services/friendbot/internal/friendbot.go]: now we just lock and unlock in one place per request.

**Comment by nullstyle** [comment]
I'm going to go ahead and merge this.  While these changes don't solve the root issues around friendbot (it still breaks pretty trivially under lots of concurrent requests) this PR should improve things somewhat while I do a round of improvements in transaction submission and testing.

**Pull request merged**